include/picongpu/particles/atomicPhysics/kernel/CheckForOverSubscription.kernel

```diff
@@ -118,18 +114,79 @@ namespace picongpu::particles::atomicPhysics::kernel
                             // but at least one average one macro ion should be rejected
                             sim.unit.typicalNumParticlesPerMacroParticle() / deltaWeight);
 
-                        // set flag that histogram has over subscribed bin
-                        alpaka::atomicExch(
-                            worker.getAcc(),
-                            &histogramOverSubscribed,
-                            u32(true),
-                            ::alpaka::hierarchy::Threads{});
+                        // set flag that we found at least one over subscribed resource
+                        sharedResourcesOverSubscribed = true;
                     }
                     else
                         rejectionProbability = -1._X;
 
-                    rejectionProbabilityCache.set(binIndex, rejectionProbability);
+                    rejectionProbabilityCache.setBin(binIndex, rejectionProbability);
                 });
+
+
+            T_FieldEnergyUseCache const& eFieldEnergyUseCache = fieldEnergyUseCacheBox(superCellFieldIdx);
+            DataSpace<picongpu::simDim> const superCellCellOffset = superCellIdx * picongpu::SuperCellSize::toRT();
+            auto forEachCell = pmacc::lockstep::makeForEach<T_FieldEnergyUseCache::numberCells, T_Worker>(worker);
+            forEachCell(
+                [&worker,
+                 &superCellCellOffset,
+                 &eFieldBox,
+                 &eFieldEnergyUseCache,
+                 &sharedResourcesOverSubscribed,
+                 &rejectionProbabilityCache](uint32_t const linearCellIndex)
+                {
+                    DataSpace<picongpu::simDim> const localCellIndex
+                        = pmacc::math::mapToND(picongpu::SuperCellSize::toRT(), static_cast<int>(linearCellIndex));
+                    DataSpace<picongpu::simDim> const cellIndex = localCellIndex + superCellCellOffset;
+
+                    // sim.unit.charge()^2 * sim.unit.time()^2 / (sim.unit.mass() * sim.unit.length()^3)
+                    //  * sim.unit.length()^3
+                    // = sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()
+                    constexpr float_X eps0HalfTimesCellVolume
+                        = (picongpu::sim.pic.getEps0() / 2._X) * picongpu::sim.pic.getCellSize().productOfComponents();
+
+                    // sim.unit.charge()^2 * sim.unit.time()^2 / sim.unit.mass()
+                    //  * ((sim.unit.mass() * sim.unit.length())/(sim.unit.time()^2 * sim.unit.charge()))^2
+                    // = sim.unit.charge()^2 * sim.unit.time()^2 * sim.unit.mass()^2 * sim.unit.length()^2
+                    //  / (sim.unit.mass() * sim.unit.time()^4 * sim.unit.charge()^2)
+                    // = sim.unit.mass() * sim.unit.length()^2/ (sim.unit.time()^2 * sim.unit.length())
+                    // sim.unit.energy()
```

Comment:
In this last line, is there a mathematical sign (equal, multiplication, division) missing?

Reply:
Btw, are these relations needed later on or why are these here?

Reply:
The last comment line is supposed to indicate the unit of the variable below, and the comments above show the derivation of the unit.

We switched documentation style since this PR, the [newer version](https://github.com/ComputationalRadiationPhysics/picongpu/blob/6355aa37a8d473cc5a37e8149c984ca9a277432a/include/picongpu/particles/atomicPhysics/kernel/CalculateRejectionProbability.hpp#L103-L106) should be easier to understand.

They are notes tracking the unit of variables to allow easier checking of unit conversions.